**Code from the chromatin variation 3d project**

Contact: oursu@stanford.edu

**Contents**

- **browserTracks** - code for creating the browser tracks displayed at http://mitra.stanford.edu/kundaje/chromovar3d

- **DNase** - code for processing the DNase data

- **GWASnetworks** - code for the GWAS network analysis 

- **MotifAnalysis** - code for computing correlation between signal and TF motif scores across individuals

- **RNAseq** - code for processing the RNAseq data

- **SNPs** - code for merging together the SNP set used in this analysis

- **peakCalling** - code for processing the histone mark data for H3K4me1, H3K4me3, H3K27ac

**More code from this paper**

For code related to the QTL calling, and HiC quantification and enrichment refer to http://www.zaugg.embl.de/data-and-tools/distal-chromatin-qtls/