
Motif analysis
==============

Here is the cross-TF analysis, where we perform the multiple testing correction.

INFILE='/srv/gsfs0/projects/snyder/oursu/histoneQTL/motif_analysis/results/2015-01-13/Local/LocalAnalysis.HMARK.gz'
SNPQTL='/srv/gsfs0/projects/snyder/oursu/histoneQTL/motif_analysis/results/2015-01-13/Local/SNPQTLmatrix/SNPQTLmatrix.HMARK.gz'

```{r,eval=F,dev='pdf'}

annotate_with_QTL=function(TFdata.mht,snpqtl.f){
	snpqtlmat=read.table(snpqtl.f,header=TRUE)
	snp_peak=paste(snpqtlmat$SNP,'_',snpqtlmat$gene,sep='')
	rownames(snpqtlmat)=snp_peak
	TFdata.snp_peak=paste(TFdata.mht$snp,'_',TFdata.mht$peak,sep='')
	common_items=intersect(as.character(snp_peak),as.character(TFdata.snp_peak))
	commons=which(TFdata.snp_peak %in% common_items)
	TFdata.mht$QTL=as.character(TFdata.mht$QTL)
	TFdata.mht[commons,'QTL']=as.character(snpqtlmat[TFdata.snp_peak[commons],'QTL'])
	TFdata.mht$QTL=as.factor(TFdata.mht$QTL)
	return (TFdata.mht)
}

process_hmark_motifData=function(hmark,INFILE,SNPQTL){
	in.f=gsub('HMARK',hmark,INFILE)
	TFdata=read.table(in.f,header=T)
	snpqtl.f=gsub('HMARK',hmark,SNPQTL)

        # - correct MHT across full dataset. Add another column
        TFdata.mht=data.frame(TFdata,crossTF_pBH.spearman=p.adjust(as.numeric(as.character(TFdata$genomeWide_match.p.sp)),method="BH"),hmark=hmark,QTL='nothing')

	#add info about whether it's a QTL/other
	TFdata.mht.qtl=annotate_with_QTL(TFdata.mht,snpqtl.f)
	return(TFdata.mht.qtl)
}

motif_datasets=list()
hmarks=c('H3K4ME1','H3K4ME3','H3K27AC','dhs','RNA')
hmarks=c('RNA','H3K4ME3')
first=T
for (hmark in hmarks){
    print(hmark)
    motif_datasets[[hmark]]=process_hmark_motifData(hmark,INFILE,SNPQTL)
    if (first!=TRUE){
       total=rbind(total,motif_datasets[[hmark]])
    }   
    if (first==TRUE){
       total=motif_datasets[[hmark]]
       first=FALSE
    }
}
write.table(total,file=paste(gsub('HMARK','',INFILE),'.CompiledDataAcrossTFs',sep=''),quote=F,row.names=F,col.names=T,sep='\t')
save(motif_datasets,file=paste(gsub('HMARK','',INFILE),'.CompiledDataAcrossTFs.rda',sep=''))
```

Now, restrict to peaks with QTLs. Barplot of number of peaks explained, grouped by the number of SNPs disrupted per peak.

```{r MotifsLocationsPerPeak,dev='pdf'}
#Get data for barplot
hmarks=c('H3K4ME1','H3K27AC','H3K4ME3','RNA','dhs')
hmarks=c('H3K4ME3','RNA')
numIntervals=2
barplot_data=data.frame(array(0,dim=c(length(hmarks),numIntervals)))
rownames(barplot_data)=hmarks
colnames(barplot_data)=c('1','2','3','>=4')
for (i in c(1:length(hmarks))){
  hmark=hmarks[i]

  #Numbers of interest
  data=motif_datasets[[hmark]]
  #Keep only qtls with sig motif disruptions
  sig_thresh=0.05
  qtl.data=data[intersect(which(!is.na(data$QTL)),which(as.numeric(as.character(data$crossTF_pBH.spearman))<=sig_thresh)),]
  #Keep only unique peak-scoredRegion combinations
  qtl.peakScoredRegion.uniq=qtl.data
  data.dupli=which(duplicated(paste(qtl.peakScoredRegion.uniq$peak,qtl.peakScoredRegion.uniq$snp)))
  if (length(data.dupli)>0){
     qtl.peakScoredRegion.uniq=qtl.peakScoredRegion.uniq[-data.dupli,]
  }
  #Count the number of scoredRegions per peak for the marks
  scoredRegions.perPeak=table(cut(table(qtl.peakScoredRegion.uniq[,'peak']),c(0:(numIntervals-1),max(table(qtl.peakScoredRegion.uniq[,'peak'])))))
  barplot_data[hmark,]=scoredRegions.perPeak
}
pdf('/home/oursu/motiftest.pdf')
par(mar=c(5, 4, 2, 0) + 2,mgp=c(4, 1, 0))
barplot_colors=gray.colors(numIntervals, start = 0, end = 1, gamma = 2.2)
barplot(t(barplot_data),space=0.3,cex.axis=2,cex.names=2,ylim=c(0,3000),ylab='Number of peaks',xlab='Histone mark',main='Number of peaks explained through motif analysis \n By number of motif-disrupting SNPs per peak\n Considering only QTL peaks',cex.lab=2,col=barplot_colors) 
legend('topright',fill=barplot_colors,legend=c('1 SNP/peak',paste(colnames(barplot_data)[2:dim(barplot_data)[2]],' SNPs/peak',sep='')),cex=1.2)
dev.off()


```
