
Motif analysis
==============

```{r,eval=F}
#Here is how to make the html file from this rmd
require(knitr)
OUTMAIN='/srv/gsfs0/projects/snyder/oursu/histoneQTL/motif_analysis/results/2015-06-13/'
resultdir=paste(OUTMAIN,'/CrossTFanalysis/figures/',sep='')
inrmd='/srv/gsfs0/projects/kundaje/users/oursu/code/chromovar3d/MotifAnalysis/figures/2015-06-13/rmd/correlatedMotifs_fig.Rmd'
system(paste('mkdir -p ',resultdir,sep=''))
.libPaths('~/devtools/R-3.0.2/library')
setwd(resultdir)
knit2html(inrmd,output=paste(resultdir,basename(inrmd),sep=''))
```

Here is the cross-TF analysis, where we perform the multiple testing correction.

```{r}
OUTMAIN='/srv/gsfs0/projects/snyder/oursu/histoneQTL/motif_analysis/results/2015-06-13/'
INFILE_LOCAL=paste(OUTMAIN,'/CrossTFanalysis/LocalAnalysis.motifMAF05.HMARK.gz.annotated.gz',sep='')
INFILE_DISTAL=paste(OUTMAIN,'/CrossTFanalysis/DistalAnalysis.DistalPeakIsHMARK.gz',sep='')
COMBPEAKSIGNAL=paste(OUTMAIN,'/Data/CombinedPeakSignal.gz',sep='')
COMBPEAKANNO=paste(OUTMAIN,'/Data/CombinedPeakAnno.gz',sep='')
```

Here we read in the data 
=========================

```{r,eval=F,dev='pdf'}
process_hmark_motifData=function(hmark,INFILE,COMBPEAKANNO,filterForQTLpeaks=TRUE,localdistal='local'){
  panno=read.table(COMBPEAKANNO)
  colnames(panno)=c('chr','start','stop','id')
  rownames(panno)=as.character(panno$id)
	in.f=gsub('HMARK',hmark,INFILE)
	TFdata.qtl=read.table(in.f,header=T)
  peakcol=which(colnames(TFdata.qtl)=='peak')
  if (length(peakcol)>0){
    colnames(TFdata.qtl)[peakcol]='gene'
  }
	colnames(TFdata.qtl)[which(colnames(TFdata.qtl)=='gene')]='affected.gene'
	#annotate peak coordinates
  affectedpeaks=as.character(TFdata.qtl$affected.gene)
  if (localdistal=='local'){
    affectedpeaks=paste(hmark,'_',as.character(TFdata.qtl$affected.gene),sep='')
    TFdata.qtl$affected.gene=affectedpeaks
  }
  panno_now=data.frame(panno[affectedpeaks,])
	TFdata.qtl=data.frame(data.frame(TFdata.qtl),
                  chr=panno_now[,'chr'],start=panno_now[,'start'],stop=panno_now[,'stop'])
  colnames(TFdata.qtl)[(ncol(TFdata.qtl)-2):ncol(TFdata.qtl)]=c('affected.peak.chr',
                                                                'affected.peak.start',
                                                                'affected.peak.end')
  if (localdistal=='distal'){
    TFdata.qtl=cbind(TFdata.qtl,
                    panno[as.character(TFdata.qtl$localpeak),1:3])
    colnames(TFdata.qtl)[(ncol(TFdata.qtl)-2):ncol(TFdata.qtl)]=c('localpeak.chr',
                                                                'localpeak.start',
                                                                'localpeak.end')
  }
  if (localdistal=='local'){
    TFdata.qtl=cbind(TFdata.qtl,local.peak.chr=NA,local.peak.start=NA,local.peak.end=NA)
  }
	if (filterForQTLpeaks==TRUE){
	   maxqtl_rows=which(as.character(TFdata.qtl$pass.pvalTH)=='pass')
	   qtl_peaks=unique(as.character(TFdata.qtl[maxqtl_rows,'affected.gene']))
	   if (length(qtl_peaks)>0){
	      TFdata.qtl=TFdata.qtl[which(as.character(TFdata.qtl$affected.gene) %in% qtl_peaks),]
	   }
	}
        # - correct MHT across full dataset. Add another column
        TFdata.qtl.mht=data.frame(TFdata.qtl,
			crossTF_pBH.spearman=p.adjust(as.numeric(as.character(TFdata.qtl$genomeWide_match.p.sp)),method="BH"),affected.hmark=hmark)
      if (localdistal=='distal'){
        TFdata.qtl.mht$affected.hmark=gsub('_[0-9]*','',TFdata.qtl.mht$affected.gene)
      }
	return(TFdata.qtl.mht)
}

browserTrack=function(datain,out){
  system(paste('mkdir ',dirname(out),sep=''))
  motifs=unique(datain$motifName)
  for (m in motifs){
    datacur=datain[which(as.character(datain$motifName)==m),]
    datacur=datacur[which(as.numeric(as.character(datacur$crossTF_pBH.spearman))<=0.05),]
    #print(colnames(datacur))
    chromo=datacur[,which(grepl('chr',colnames(datacur)))[1]]
    snppos=datacur[,which(grepl('snp.pos',colnames(datacur)))[1]]
    data=paste('chr',chromo,':',snppos-1,'-',snppos,
		'\tchr',chromo,':',datacur$affected.peak.start,'-',datacur$affected.peak.end,
		'\t',datacur[,which(grepl('spearmanCorr_MotifScoreVS',colnames(datacur)))[1]],sep='')
	 write.table(data,file=paste(out,m,'pairwise.track',sep=''),sep='\t',quote=F,row.names=F,col.names=F)
  }
}

bedfile_correlated_motifs=function(datain,out,sig_threshold){
  #one with all variable motifs
  varmotifs=paste(dirname(out),'/VariableMotifs',sep='')
  cormotifs=paste(dirname(out),'/CorrelatedMotifs',sep='')
  #one with just significant ones
  system(paste('mkdir -p ',varmotifs,sep=''))
  system(paste('mkdir -p ',cormotifs,sep=''))
  #get column names correct
  CHR_COL=which(grepl('chr',colnames(datain)))[1]
  SNPPOS_COL=which(grepl('snp.pos',colnames(datain)))[1]
  motifs=as.character(unique(datain$motifName))
  for (m in motifs){
    print(m)
    datacur=datain[which(as.character(datain$motifName)==m),]
    datacur_sig=datacur[which(as.numeric(as.character(datacur$crossTF_pBH.spearman))<=sig_threshold),]
    write_var=unique(paste(datacur[,CHR_COL],as.numeric(datacur[,SNPPOS_COL])-1,as.numeric(as.character(datacur[,SNPPOS_COL])),m,sep='\t'))
    write_sig=unique(paste(datacur_sig[,CHR_COL],as.numeric(datacur_sig[,SNPPOS_COL])-1,as.numeric(as.character(datacur_sig[,SNPPOS_COL])),m,sep='\t'))
    write.table(write_var,file=paste(varmotifs,'/','varMotifs.',m,sep=''),
      quote=F,row.names=F,col.names=F,sep='\t')
    system(paste('gzip ',paste(varmotifs,'/','varMotifs.',m,sep=''),sep=''))
    write.table(write_sig,file=paste(cormotifs,'/','correlatedMotifs.',m,sep=''),
      quote=F,row.names=F,col.names=F,sep='\t')
    system(paste('gzip ',paste(cormotifs,'/','correlatedMotifs.',m,sep=''),sep=''))
  }
  #also write some files with all motifs combined 
  hmarks_affected=unique(as.character(datain$affected.hmark))
  for (hmark in hmarks_affected){
    datacur=datain[which(as.character(datain$affected.hmark)==hmark),]
    datacur_sig=datacur[which(as.numeric(as.character(datacur$crossTF_pBH.spearman))<=sig_threshold),]
    print(summary(datacur_sig))
    write_var=unique(paste(datacur[,CHR_COL],as.numeric(datacur[,SNPPOS_COL])-1,as.numeric(as.character(datacur[,SNPPOS_COL])),datacur$motifName,sep='\t'))
    write_sig=unique(paste(datacur_sig[,CHR_COL],as.numeric(datacur_sig[,SNPPOS_COL])-1,as.numeric(as.character(datacur_sig[,SNPPOS_COL])),datacur_sig$motifName,sep='\t'))
    write.table(write_var,file=paste(varmotifs,'/','varMotifs.',hmark,'.allTFs',sep=''),
        quote=F,row.names=F,col.names=F,sep='\t')
    system(paste('gzip ',paste(varmotifs,'/','varMotifs.',hmark,'.allTFs',sep=''),sep=''))
    write.table(write_sig,file=paste(cormotifs,'/','correlatedMotifs.',hmark,'.allTFs',sep=''),
        quote=F,row.names=F,col.names=F,sep='\t')
    system(paste('gzip ',paste(cormotifs,'/','correlatedMotifs.',hmark,'.allTFs',sep=''),sep=''))
  }

}

#aggregate data
motif_datasets_local=list()
motif_datasets_distal=list()
motif_datasets_local_all=list()
hmarks=c('H3K4ME1','H3K4ME3','H3K27AC','dhs','RNA')
first=TRUE
for (hmark in hmarks){
    print(hmark)
    motif_datasets_local[[hmark]]=process_hmark_motifData(hmark,INFILE_LOCAL,COMBPEAKANNO,TRUE,'local')
    motif_datasets_local_all[[hmark]]=process_hmark_motifData(hmark,INFILE_LOCAL,COMBPEAKANNO,FALSE,'local')
    print('done local')
    motif_datasets_distal[[hmark]]=process_hmark_motifData(hmark,INFILE_DISTAL,COMBPEAKANNO,FALSE,'distal')
    if (first!=TRUE){
       total_local=rbind(total_local,motif_datasets_local[[hmark]])
       total_distal=rbind(total_distal,motif_datasets_distal[[hmark]])
       total_local_full=rbind(total_local_full,motif_datasets_local_all[[hmark]])
    }   
    if (first==TRUE){
       total_local=motif_datasets_local[[hmark]]
       total_distal=motif_datasets_distal[[hmark]]
       total_local_full=motif_datasets_local_all[[hmark]]
       first=FALSE
    }
}
write.table(total_local_full,file=paste(gsub('HMARK','',INFILE_LOCAL),'.AllPeaks.CompiledDataAcrossTFs',sep=''),quote=F,row.names=F,col.names=T,sep='\t')
save(motif_datasets_local_all,file=paste(gsub('HMARK','',INFILE_LOCAL),'.AllPeaks.CompiledDataAcrossTFs.rda',sep=''))
write.table(total_local,file=paste(gsub('HMARK','',INFILE_LOCAL),'.CompiledDataAcrossTFs',sep=''),quote=F,row.names=F,col.names=T,sep='\t')
save(motif_datasets_local,file=paste(gsub('HMARK','',INFILE_LOCAL),'.CompiledDataAcrossTFs.rda',sep=''))
write.table(total_distal,file=paste(gsub('HMARK','',INFILE_DISTAL),'.CompiledDataAcrossTFs',sep=''),quote=F,row.names=F,col.names=T,sep='\t')
save(motif_datasets_distal,file=paste(gsub('HMARK','',INFILE_DISTAL),'.CompiledDataAcrossTFs.rda',sep=''))
#TODO: run these
browserTrack(total_local_all,paste(dirname(INFILE_LOCAL),'/LocalMotifTracks/motifCorr',sep=''))
browserTrack(total_distal,paste(dirname(INFILE_DISTAL),'/DistalMotifTracks/motifCorr',sep=''))
bedfile_correlated_motifs(total_local_all,paste(dirname(INFILE_LOCAL),'/SNPsInMotifsBed/Local/_',sep=''),0.05)
bedfile_correlated_motifs(total_distal,paste(dirname(INFILE_DISTAL),'/SNPsInMotifsBed/Distal/_',sep=''),0.05)
```

Functions to make figures
==========================

```{r MotifsLocationsPerPeak,dev='pdf'}
#FRACTIONS PLOT
#==============
fractionsPlot=function(data,SIG_THRESH,out,hmark,N_MIN,keep){
  colnames(data)=gsub('peak','gene',colnames(data))
  data.disrupt=data[which(as.numeric(as.character(data$crossTF_pBH.spearman))<=SIG_THRESH),]
  data.dupli=which(duplicated(paste(data.disrupt$motifName,data.disrupt$snp,data.disrupt$affected.gene)))
  if (length(data.dupli)>0){
    data.disrupt=data.disrupt[-data.dupli,]
    data=data[-data.dupli,]
  }
  motif_counts=table(data$motifName)
  motif_counts.disrupt=table(data.disrupt$motifName)
  motif_counts.disrupt.neg=table(data.disrupt[which(as.numeric(as.character(data.disrupt$spearmanCorr_MotifScoreVSgene))<0),
    'motifName'])
  motif_counts.disrupt.pos=table(data.disrupt[which(as.numeric(as.character(data.disrupt$spearmanCorr_MotifScoreVSgene))>=0),
    'motifName'])
  motifs_total=as.character(unique(data.disrupt$motifName))
  percent_disrupted=data.frame(array(0,dim=c(length(motifs_total),5)))
  rownames(percent_disrupted)=motifs_total
  colnames(percent_disrupted)=c('PercentDisrupted','N','Positive','Negative','TF')
  percent_disrupted[names(motif_counts.disrupt),1]=(motif_counts.disrupt)/(motif_counts[names(motif_counts.disrupt)])
  percent_disrupted[,'N']=motif_counts[rownames(percent_disrupted)]
  percent_disrupted[names(motif_counts.disrupt.pos),'Positive']=100*(motif_counts.disrupt.pos)/(motif_counts[names(motif_counts.disrupt)])
  percent_disrupted[names(motif_counts.disrupt.pos),'Negative']=100*(motif_counts.disrupt.neg)/(motif_counts[names(motif_counts.disrupt)])
  percent_disrupted$TF=paste(gsub('motif.pouya.Motif.','',
                    gsub('scanThresh0','',rownames(percent_disrupted))),'.  N=',percent_disrupted$N,sep='')
  print(hmark)
  print(percent_disrupted)
  percent_disrupted=percent_disrupted[which(as.numeric(as.character(percent_disrupted$N))>=N_MIN),]
  percent_disrupted$TF=factor(percent_disrupted$TF,
                    levels=percent_disrupted$TF[order(as.numeric(as.character(percent_disrupted$Positive))-as.numeric(as.character(percent_disrupted$Negative)))])
  percent_disrupted=percent_disrupted[order(percent_disrupted$PercentDisrupted,decreasing=TRUE),]
  #percent_disrupted=percent_disrupted[c(1:keep),]
  require(reshape2)
  percent_disrupted.melt=melt(percent_disrupted[,c('TF','Negative','Positive')])
  colnames(percent_disrupted.melt)=c('TF','CorrSign','value')
  percent_disrupted.melt$CorrSign=factor(percent_disrupted.melt$CorrSign,levels=c('Negative','Positive'))
  #pdf(paste(out,hmark,'.pdf',sep=''))
  require(ggplot2)
  print(ggplot(percent_disrupted.melt,aes(x=TF,y=value,fill=CorrSign))+guides(fill=guide_legend(title='Correlation sign'))+scale_fill_manual(values=c('black','lightblue'))+geom_bar(stat="identity")+coord_flip() +theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))+ylab('Percent tested motif-peak instances \nthat correlate with histone mark')+xlab('TF motif')+ggtitle(hmark))
  #dev.off()
  df=percent_disrupted[,c('TF','Negative','Positive')]
  df$Negative=-df$Negative
  require(lattice)
  bc = barchart(TF~Negative+Positive, data = df, stack=TRUE)
  ## set the plotting parameters
  Tset = trellis.par.get()
  Tset$superpose.polygon$col = c("black","lightblue")
  trellis.par.set(Tset)
  print(plot(bc,main=hmark))
  return(cbind(percent_disrupted.melt,hmark=hmark))
}

```

Make figures
=============
```{r}
SIG_THRESH=0.05
MIN_N=50
#Get data
#=========
load(paste(gsub('HMARK','',INFILE_LOCAL),'.AllPeaks.CompiledDataAcrossTFs.rda',sep=''))
load(paste(gsub('HMARK','',INFILE_DISTAL),'.CompiledDataAcrossTFs.rda',sep=''))
total_local=read.table(paste(gsub('HMARK','',INFILE_LOCAL),'.CompiledDataAcrossTFs',sep=''),header=TRUE)
total_distal=read.table(paste(gsub('HMARK','',INFILE_DISTAL),'.CompiledDataAcrossTFs',sep=''),header=TRUE)
total_local_all=read.table(paste(gsub('HMARK','',INFILE_LOCAL),'.AllPeaks.CompiledDataAcrossTFs',sep=''),header=TRUE)
```

#Barplots of counts
#===================
```{r MotifsCountsPerPeak,dev='pdf',fig.width=7,fig.height=4}
#total=rbind(cbind(total_local[,c('snp','affected.gene','affected.hmark')],type='local'),cbind(total_distal[,c('snp','affected.gene','affected.hmark')],type='distal'))
#dataset=total_local
get_counts=function(dataset,max_n,SIG_THRESH){
  data.res=dataset[which(as.numeric(as.character(dataset$crossTF_pBH.spearman))<=SIG_THRESH),c('snp','affected.gene','affected.hmark')]
  hmarks=c('H3K4ME1','H3K4ME3','H3K27AC','dhs','RNA')
  counts_matrix=data.frame(hmark=rep(hmarks,each=max_n),numSNPs=rep(c(1:max_n),times=length(hmarks)),
    countSNPs=0)
  rownames(counts_matrix)=paste(counts_matrix$hmark,counts_matrix$numSNPs)
  for (hmark in hmarks){
    data.hmark=data.res[which(as.character(data.res$affected.hmark)==hmark),]
    #first, we remove duplicate snp-peak pairs
    dupli.snppeak=which(duplicated(data.hmark[,c('snp','affected.gene')]))
    if (length(dupli.snppeak)>0){
      data.hmark=data.hmark[-dupli.snppeak,]
    }
    #now, count each peak
    counts_per_peak=table(as.character(data.hmark[,'affected.gene']))
    counts_per_peak[counts_per_peak>max_n]=max_n
    sum_counts=table(counts_per_peak)
    counts_matrix[paste(hmark,names(sum_counts)),'countSNPs']=sum_counts
  }
  rownames(counts_matrix)=NULL
  return(counts_matrix)
}
#motif_counts=rbind(cbind(get_counts(total_local,4,0.05),typeCor='Local motif disruptions'),
#            cbind(get_counts(total_distal,4,0.05),typeCor='Distal motif disruptions'))
motif_counts=rbind(cbind(get_counts(total_local_all[which(total_local_all$pass.pvalTH=='pass'),],4,0.05),typeCor='Local motif disruptions'),
            cbind(get_counts(total_distal,4,0.05),typeCor='Distal motif disruptions'))
print('total numbers')
print(cbind(get_counts(total_local_all,1,Inf),typeCor='local'))
print('total tested peaks with QTL')
print(cbind(get_counts(total_local,1,Inf),typeCor='local'))
print('total tested peaks with distal QTL')
print(cbind(get_counts(total_distal,1,Inf),typeCor='distal'))
print('correlated numbers')
print(rbind(cbind(get_counts(total_local,1,0.05),typeCor='local'),
            cbind(get_counts(total_distal,1,0.05),typeCor='distal')))
print(' all tested locals')
print(get_counts(total_local_all[which(total_local_all$pass.pvalTH=='pass'),],1,Inf))
print('sig after correction using all tested locals')
print(get_counts(total_local_all[which(total_local_all$pass.pvalTH=='pass'),],1,0.05))
motif_counts$hmark=factor(motif_counts$hmark,levels=c('RNA','dhs','H3K27AC','H3K4ME3','H3K4ME1'))
motif_counts$numSNPs=as.character(motif_counts$numSNPs)
motif_counts[which(as.character(motif_counts$numSNPs)=='1'),'numSNPs']='1 SNP per peak'
motif_counts[which(as.character(motif_counts$numSNPs)=='2'),'numSNPs']='2 SNPs per peak'
motif_counts[which(as.character(motif_counts$numSNPs)=='3'),'numSNPs']='3 SNPs per peak'
motif_counts[which(as.character(motif_counts$numSNPs)=='4'),'numSNPs']='>=4 SNPs per peak'
motif_counts$numSNPs=factor(motif_counts$numSNPs, levels=c('1 SNP per peak','2 SNPs per peak',
  '3 SNPs per peak','>=4 SNPs per peak'))
require(ggplot2)
pdf(paste(OUTMAIN,'/CrossTFanalysis/figures/motifCorrCount.pdf',sep=''),
  width=8,height=4)
print(ggplot(motif_counts,aes(x=hmark,y=countSNPs,fill=numSNPs))+scale_fill_manual(values=c(1,2,7,4))+facet_wrap(~typeCor,ncol=1)+geom_bar(stat='identity')+coord_flip() +theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"))+guides(fill=guide_legend(title=NULL))+ylab('Number of peaks with motif disruption')+xlab('')+
theme(text = element_text(size=15)))
dev.off()
```
```{r FractionPlotLocal,dev='pdf',fig.width=6,fig.height=8}
#Fraction plots
#==============
hmarks=c('H3K4ME1','H3K4ME3','H3K27AC','dhs','RNA')
first=TRUE
for (hmark in hmarks){
  fractionsPlot(motif_datasets_local_all[[hmark]],SIG_THRESH,paste(INFILE_LOCAL,'FractionPlot',hmark,sep=''),hmark,MIN_N,10)
}
```

```{r FractionPlotDistal,dev='pdf',fig.width=6,fig.height=10}
hmarks=c('H3K4ME1','H3K4ME3','H3K27AC','dhs','RNA')
for (hmark in hmarks){
  fractionsPlot(total_distal[which(as.character(total_distal$affected.hmark)==hmark),],SIG_THRESH,paste(INFILE_DISTAL,'FractionPlot',hmark,sep=''),hmark,MIN_N,10)
}
```

```{r, eval=FALSE}
#local-distal correlation plot
locals_of_interest=paste(total_distal$snp,total_distal$localpeak,total_distal$motifName)
#rownames(total_local)=paste(total_local$SNP,total_local$affected.gene,total_local$motifName)
rownames(total_local_all)=paste(total_local_all$SNP,total_local_all$affected.gene,total_local_all$motifName)
#a=setdiff(locals_of_interest,unique(rownames(total_local)))
a=setdiff(locals_of_interest,unique(rownames(total_local_all)))
print('Locals tested for distal but not for local: (should be empty)')
print(a)
commons=intersect(unique(locals_of_interest),unique(rownames(total_local_all)))
distal=cbind(total_distal,localCorr=total_local_all[locals_of_interest,'spearmanCorr_MotifScoreVSpeak'],
  BHlocal=total_local_all[locals_of_interest,'crossTF_pBH.spearman'])
distal2=cbind(distal,sig_local=(as.numeric(as.character(distal$BHlocal))<=SIG_THRESH),
  sig_distal=(as.numeric(as.character(distal$crossTF_pBH.spearman))<=SIG_THRESH),
  sig=((as.numeric(as.character(distal$BHlocal))<=SIG_THRESH)*(as.numeric(as.character(distal$crossTF_pBH.spearman))<=SIG_THRESH)>0))

pdf(paste(OUTMAIN,'/CrossTFanalysis/figures/localDistalCorr.pdf',sep=''))
motifs=unique(distal$motifName)
for (m in motifs){
  curdata=distal2[which(as.character(distal2$motifName)==m),]
  sig_local=which(as.numeric(as.character(curdata$BHlocal))<=SIG_THRESH)
  sig_distal=which(as.numeric(as.character(distal$BHlocal))<=SIG_THRESH)
  sigs=union(sig_local,sig_distal)
  distal2
  plot(curdata$localCorr,curdata$spearmanCorr_MotifScoreVSpeak,
    main=gsub('motif.pouya.Motif.','',m),col=1+curdata$sig_local,
    bg=3+curdata$sig_distal,
    xlab='Spearman correlation between motif and local peak',
    ylab='Spearman correlation between motif and distal peak',
    xlim=c(-1,1),ylim=c(-1,1),pch=21)
}
dev.off()
```


