<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="X-Frame-Options" content="allow">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <link rel="stylesheet" type="text/css" href="http://epigenomegateway.wustl.edu/browser/style.css" />
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/base.js"></script>
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/personality.js"></script>
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/embed.js"></script>

    <link rel="icon" href="../../favicon.ico">
    <link href="./css/bootstrap.min.css" rel="stylesheet">
    <link href="./css/bootstrap-theme.min.css" rel="stylesheet">

    <script src="../../assets/js/ie-emulation-modes-warning.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.0.2/jquery.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
    
    <link href="./css/jquery.dataTables.css" rel="stylesheet">

<!--    <script src="../../assets/js/ie-emulation-modes-warning.js"></script>-->
    <script type="text/javascript" src="./js/jquery.tablehover.min.js"></script>
    <script type="text/javascript" src="./js/jquery.dataTables.min.js"></script>
    <script type="text/javascript" src="./js/jquery.dataTables.rowGrouping.js"></script>

    <script type="text/javascript" src="./js/jquery.ui.selectableTable.js"></script>

    <link href="theme.css" rel="stylesheet">


    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <script type="text/javascript" src="./data.js"></script>
    <script type="text/javascript" src="./func.js"></script>

    <script type="text/javascript">
	$(function(){
		$("#header").load("header.html");
		$("#footer").load("footer.html");
	});

	$(document).ready(function( $ ) { 
		//alert('document is ready');
		$("div_vis_area").hide();
	});
	

    </script>

    <title>Chromovar3d</title>
  </head>


  <body role="document">
    <div id="header"></div>

    <div class="container-fluid theme-showcase" role="main" id="body">

<h2>Abstract</h2>

<table><tbody>
<td valign="top">
<p>
Deciphering the impact of genetic variants on gene regulation is fundamental to understanding human disease. Although gene regulation often involves long-range interactions, it is unknown to what extent non-coding genetic variants influence distal molecular phenotypes. Here, we integrate chromatin profiling for three histone marks in lymphoblastoid cell lines (LCLs) from 75 sequenced individuals with LCL-specific Hi-C and ChIA-PET-based chromatin contact maps to uncover one of the largest collections of local and distal histone quantitative trait loci (hQTLs). Distal QTLs are enriched within topologically associated domains and exhibit largely concordant variation of chromatin state coordinated by proximal and distal non-coding genetic variants. Histone QTLs are enriched for common variants associated with autoimmune diseases and enable identification of putative target genes of disease-associated variants from genome-wide association studies. These analyses provide insights into how genetic variation can affect human disease phenotypes by coordinated changes in chromatin at interacting regulatory elements.         
</p>

<br><br><h2 style="margin-top:0; padding-top:0">Visualization</h2>

</td>
<td><img src="./images/webserverpicture.png" width=300></img></td>
</tbody></table>

            <div class="well">
               <h2><img src="./images/db.png" width="20"/> DATA SOURCE</h2>

               <ul>
                <li>Download URL for browser tracks: <a href="http://chromovar3d.stanford.edu/browserTracks/" target="_blank">http://chromovar3d.stanford.edu/browserTracks/</a></li>
               </ul>
                <br>
                <select class="ddl_assay" id="ddl_assay_bigwig" onchange="onchange_ddl_assay_bigwig(this);">
                    <option>Choose a chromatin mark</option>
                </select> &nbsp &nbsp
                <button id="btn_bigwig" onclick="onclick_btn_bigwig(this);" onclick="init_ddl_bigwig()"> Visualize all histones and DNase </button>                
                <br> <input type="checkbox" id="chkbox_show_interaction_local" onclick="init_ddl_bigwig()" checked> Show local QTLs</input>
                <br> <input type="checkbox" id="chkbox_show_interaction_distal" onclick="init_ddl_bigwig()" checked> Show distal QTLs</input>
                <br> <input type="checkbox" id="chkbox_show_signal_tracks" onclick="init_ddl_bigwig()"> Show signal tracks</input>
                <br> <input type="checkbox" id="chkbox_show_interaction_hic" onclick="init_ddl_bigwig()"> Show HiC (takes several minutes to visualize according to your computer resources)</input>
                <br> <input type="checkbox" id="chkbox_show_interaction_chiapet" onclick="init_ddl_bigwig()"> Show ChiA-PET</input>
                <br> <input type="checkbox" id="chkbox_new_page_bigwig" onclick="init_ddl_bigwig()" checked> Open in a new page (deactivate pop-up blockers)</input>
                <br><br>
                <p><b>Legend for QTL tracks</b>: The QTL tracks contain all associations passing the p-value threshold corresponding to a 10% FDR. Each association is represented in the "interaction" format, that is, as a link between the QTL SNP (1 base) and the peak. The scores (colors of the links) represent the association effect sizes.</p>
                <br><div style='margin:20px;border:solid 2px green;display:none;' id='embed_bigwig'></div>
            </div>
            


<h2>Data overview</h2>
<p>
Our data are organized in the following directories:
</p>
<pre>
Data processing: Personal genomes, Histone marks, DNaseI-seq, RNAseq, HiC, ChIA-PET

QTL analysis: QTLs

Other analyses: Motif analysis/TF binding, GWAS

Raw data: GEO

Code: Code
</pre>

<h2>Quick start</h2>

<p>
The most common use of this portal is using the QTL analysis.
For that, all you need to download is the QTL directory at <a href="http://chromovar3d.stanford.edu/QTLs/" target="_blank">http://chromovar3d.stanford.edu/QTLs/</a>
</p>

<a href="http://chromovar3d.stanford.edu/QTLs/results" target="_blank">http://chromovar3d.stanford.edu/QTLs/results</a> has the following contents (see READMEs in each of these for more detailed descriptions of the files):

<h3>Genomic regions</h3>
<ul>
<li>
QTLs/peak_info - Files containing information about the peaks used in this study. Specifically, these files assign peak IDs to peaks in this study, and these peak IDs are referenced in the localQTL and distalQTL analysis. These peaks and genes represent the rows in the data matrices below. The peak IDs follow the order or peaks (or genes) in the data matrices from the subdirectory uncorrectedSignal. There are 2 subdirectories:
</li>
<li>
QTLs/peak_info/total_peaks - All genomic regions (peaks or genes) for which we have computed data matrices for QTL calling. These data matrices are in QTLs/results/uncorrrectedSignal/. Each peak is associated with an ID that is referenced in our analyses below (for instance, the row names of the PEER-corrected data matrices in QTLs/results/correctedSignal/ are the peak IDs.
</li>
<li>
QTLs/peak_info/tested_peaks - Genomic regions (peaks or genes) that were used in the QTL calling. They are fewer than the total regions, because we used a filter on peak variability for testing it for being a QTL (see Methods). We provide raw signal data matrices for the complete set in total_peaks, so you can start from these and perform your analysis, if you wish to not impose the variability filter (our filter on variability might not be the best filter).
</li>
</ul>

<h3>Data matrices</h3>
<ul>
<li>
QTLs/uncorrectedSignal - the initial data matrices, obtained by computing the mean signal inside histone mark peaks, DNaseI(dhs) peaks or gene expression estimates.
</li>
<li>
QTLs/correctedSignal - the same matrices as in uncorrectedSignal, but after removing confounding factors using PEER. For the histone marks, the data matrices in uncorrectedSignal and correctedSignal have the same dimensions. For dhs, the data matrix in uncorrectedSignal contains the signal in all 700k dhs merged peaks we called. However, the data matrix in correctedSignal only contains signal from the top 200k peaks (ranked by signal intensity). For RNA, the data matrix from uncorrectedSignal contains signal in all protein-coding and lncRNA genes used in this study, whereas the data matrix in correctedSignal only contains signal for those genes that did not have 0 entries in the uncorrected matrix. 
</li>
<li>To get the coordinates of the genomic regions in the correctedSignal matrices, use the first column, which is the peak/gene ID, as referenced in peak_info/total_peaks/</li>
</ul>

<h3>QTLs</h3>
<ul>
<li>QTLs/localQTL - Bedpe files with the results from the local QTL analysis</li>
<li>QTLs/distalQTL - Bedpe files with the results from the distal QTL analysis</li>
</ul>


        <script>
            init_ddl_bigwig();
        </script>

	</div>

    <div id="footer"></div>
  </body>
</html>


