<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta http-equiv="X-Frame-Options" content="allow">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <link rel="stylesheet" type="text/css" href="http://epigenomegateway.wustl.edu/browser/style.css" />
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/base.js"></script>
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/personality.js"></script>
    <script type="text/javascript" src="http://epigenomegateway.wustl.edu/browser/js/embed.js"></script>

    <link rel="icon" href="../../favicon.ico">
    <link href="./css/bootstrap.min.css" rel="stylesheet">
    <link href="./css/bootstrap-theme.min.css" rel="stylesheet">

    <script src="../../assets/js/ie-emulation-modes-warning.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.0.2/jquery.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.10.3/jquery-ui.min.js"></script>
    
    <link href="./css/jquery.dataTables.css" rel="stylesheet">

<!--    <script src="../../assets/js/ie-emulation-modes-warning.js"></script>-->
    <script type="text/javascript" src="./js/jquery.tablehover.min.js"></script>
    <script type="text/javascript" src="./js/jquery.dataTables.min.js"></script>
    <script type="text/javascript" src="./js/jquery.dataTables.rowGrouping.js"></script>

    <script type="text/javascript" src="./js/jquery.ui.selectableTable.js"></script>

    <link href="theme.css" rel="stylesheet">


    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <script type="text/javascript" src="./data.js"></script>
    <script type="text/javascript" src="./func.js"></script>

    <script type="text/javascript">
	$(function(){
		$("#header").load("header.html");
		$("#footer").load("footer.html");
	});

	$(document).ready(function( $ ) { 
		//alert('document is ready');
		$("div_vis_area").hide();
	});
	

    </script>

    <title>Chromovar3d</title>
  </head>


  <body role="document">
    <div id="header"></div>

    <div class="container-fluid theme-showcase" role="main" id="body">

      <div class="page-header">

<h2>Abstract</h2>

<table><tbody>
<td>
<p>
Deciphering the impact of genetic variants on gene regulation is fundamental to understanding human disease. Although gene regulation often involves long-range interactions, it is unknown to what extent non-coding genetic variants influence distal molecular phenotypes. Here, we integrate chromatin profiling for three histone marks in lymphoblastoid cell lines (LCLs) from 75 sequenced individuals with LCL-specific Hi-C and ChIA-PET-based chromatin contact maps to uncover one of the largest collections of local and distal histone quantitative trait loci (hQTLs). Distal QTLs are enriched within topologically associated domains and exhibit largely concordant variation of chromatin state coordinated by proximal and distal non-coding genetic variants. Histone QTLs are enriched for common variants associated with autoimmune diseases and enable identification of putative target genes of disease-associated variants from genome-wide association studies. These analyses provide insights into how genetic variation can affect human disease phenotypes by coordinated changes in chromatin at interacting regulatory elements.         
</p>
</td>
<td><img src="./images/webserverpicture.png" width=300></img></td>
</tbody></table>

<h3>AUTHORS</h3>
<p>
Fabian Grubert <sup>1,7</sup>, Judith B. Zaugg<sup>1,2,7</sup>, Maya Kasowski<sup>1,7</sup>, Oana Ursu<sup>1,7</sup>, Damek V. Spacek<sup>1</sup>, Alicia R. Martin<sup>1</sup>, Peyton Greenside<sup>3</sup>, Rohith Srivas<sup>1</sup>, Doug H. Phanstiel<sup>1</sup>, Aleksandra Pekowska<sup>2</sup>, Nastaran Heidari<sup>1</sup>, Ghia Euskirchen<sup>1</sup>, Wolfgang Huber<sup>2</sup>, Jonathan K. Pritchard<sup>1,4,5</sup>, Carlos D. Bustamante<sup>1</sup>, Lars M. Steinmetz<sup>1,2</sup>, Anshul Kundaje<sup>1,6</sup>, and Michael Snyder<sup>1,*</sup>
</p>

<h3>AFFILIATIONS</h3>
<ul style="list-style-type: none">
<li>1) Department of Genetics, Stanford University School of Medicine, Stanford, CA 94305, USA</li>
<li>2) The European Molecular Biology Laboratory Heidelberg, 69117 Heidelberg, Germany</li>
<li>3) Biomedical Informatics Graduate Training Program, Stanford University School of Medicine, Stanford, CA 94305, USA</li>
<li>4) Department of Biology, Stanford University, Stanford, CA 94305, USA</li>
<li>5) Howard Hughes Medical Institute, Stanford University, Stanford, CA 94305, USA</li>
<li>6) Department of Computer Science, Stanford University, Stanford, CA 94305, USA</li>
<li>7) Co-first author</li>
<li>*) Correspondence: mpsnyder@stanford.edu</li>
</ul>
<p>
</p>



        <h2>Visualization</h2>

            <div class="well">
               <h2><img src="./images/db.png" width="20"/> DATA SOURCE</h2>

               <ul>
                <li>Code used for this analysis is at: <a href="https://github.com/kundajelab/chromovar3d" target="_blank">https://github.com/kundajelab/chromovar3d</a></li>
               </ul>
                <br>
                <select class="ddl_assay" id="ddl_assay_bigwig" onchange="onchange_ddl_assay_bigwig(this);">
                    <option>Choose a chromatin mark</option>
                </select> &nbsp &nbsp
                <button id="btn_bigwig" onclick="onclick_btn_bigwig(this);" onclick="init_ddl_bigwig()"> Visualize all histones and DNase </button>                
                <br> <input type="checkbox" id="chkbox_show_interaction_local" onclick="init_ddl_bigwig()" checked> Show local QTLs</input>
                <br> <input type="checkbox" id="chkbox_show_interaction_distal" onclick="init_ddl_bigwig()" checked> Show distal QTLs</input>
                <br> <input type="checkbox" id="chkbox_show_signal_tracks" onclick="init_ddl_bigwig()"> Show signal tracks</input>
                <br> <input type="checkbox" id="chkbox_show_interaction_hic" onclick="init_ddl_bigwig()"> Show HiC (takes several minutes to visualize according to your computer resources)</input>
                <br> <input type="checkbox" id="chkbox_show_interaction_chiapet" onclick="init_ddl_bigwig()"> Show ChiA-PET</input>
                <br> <input type="checkbox" id="chkbox_new_page_bigwig" onclick="init_ddl_bigwig()" checked> Open in a new page (deactivate pop-up blockers)</input>
                <br><br>
                <p><b>Legend for QTL tracks</b>: The QTL tracks contain all associations passing the p-value threshold corresponding to a 10% FDR. Each association is represented in the "interaction" format, that is, as a link between the QTL SNP (1 base) and the peak. The scores (colors of the links) represent the association effect sizes.</p>
                <br><div style='margin:20px;border:solid 2px green;display:none;' id='embed_bigwig'></div>
            </div>
            

        <script>
            init_ddl_bigwig();
        </script>

	</div>
    </div>
    <div id="footer"></div>
  </body>
</html>


